### ISSUE TITLE
buildPlugin fails in Windows (gradle 6.2)

### ISSUE THREAD

**Issue by alexbudarov** [JetBrains/gradle-intellij-plugin#487]
gradle-intellij-plugin version: 0.4.18, also tried 0.4.20.
OS: Windows
Gradle: 6.2.1, also tried 6.2.2, 6.3, 6.4

"gradlew buildPlugin" fails with the following exception:
<details>
 <summary>Click to expand</summary>
 
```java
[20:46:31][Gradle failure report] org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':studio-intellij:buildPlugin'.
[20:46:31][Gradle failure report] 	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:38)
[20:46:31][Gradle failure report] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
[20:46:31][Gradle failure report] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
[20:46:31][Gradle failure report] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
[20:46:31][Gradle failure report] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
[20:46:31][Gradle failure report] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
[20:46:31][Gradle failure report] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
[20:46:31][Gradle failure report] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
[20:46:31][Gradle failure report] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
[20:46:31][Gradle failure report] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
[20:46:31][Gradle failure report] 	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
[20:46:31][Gradle failure report] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
[20:46:31][Gradle failure report] 	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:41)
[20:46:31][Gradle failure report] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:372)
[20:46:31][Gradle failure report] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:359)
[20:46:31][Gradle failure report] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:352)
[20:46:31][Gradle failure report] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:338)
[20:46:31][Gradle failure report] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
[20:46:31][Gradle failure report] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
[20:46:31][Gradle failure report] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
[20:46:31][Gradle failure report] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
[20:46:31][Gradle failure report] 	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
[20:46:31][Gradle failure report] 	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
[20:46:31][Gradle failure report] 	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
[20:46:31][Gradle failure report] Caused by: java.nio.file.InvalidPathException: Illegal char <:> at index 59: D:\BuildAgent\work\3534af43999e68d2\modules\intellij\task ':studio-intellij:jarSearchableOptions' property 'destinationDirectory'\task ':studio-intellij:jarSearchableOptions' property 'archiveFileName'
[20:46:31][Gradle failure report] 	at org.gradle.internal.snapshot.impl.DirectorySnapshotter.snapshot(DirectorySnapshotter.java:65)
[20:46:31][Gradle failure report] 	at org.gradle.internal.vfs.impl.DefaultVirtualFileSystem.lambda$read$9(DefaultVirtualFileSystem.java:116)
[20:46:31][Gradle failure report] 	at org.gradle.internal.vfs.impl.DefaultVirtualFileSystem.lambda$read$10(DefaultVirtualFileSystem.java:114)
[20:46:31][Gradle failure report] 	at org.gradle.internal.vfs.impl.DefaultVirtualFileSystem$StripedProducerGuard.guardByKey(DefaultVirtualFileSystem.java:193)
[20:46:31][Gradle failure report] 	at org.gradle.internal.vfs.impl.DefaultVirtualFileSystem.lambda$read$11(DefaultVirtualFileSystem.java:111)
[20:46:31][Gradle failure report] 	at org.gradle.internal.vfs.impl.DefaultVirtualFileSystem.read(DefaultVirtualFileSystem.java:111)
[20:46:31][Gradle failure report] 	at org.gradle.internal.vfs.RoutingVirtualFileSystem.read(RoutingVirtualFileSystem.java:59)
[20:46:31][Gradle failure report] 	at org.gradle.internal.fingerprint.impl.DefaultFileCollectionSnapshotter$SnapshottingVisitor.visitFileTree(DefaultFileCollectionSnapshotter.java:71)
[20:46:31][Gradle failure report] 	at org.gradle.api.internal.file.collections.FileTreeAdapter.visitStructure(FileTreeAdapter.java:131)
[20:46:31][Gradle failure report] 	at org.gradle.api.internal.file.CompositeFileCollection.visitStructure(CompositeFileCollection.java:181)
[20:46:31][Gradle failure report] 	at org.gradle.internal.fingerprint.impl.DefaultFileCollectionSnapshotter.snapshot(DefaultFileCollectionSnapshotter.java:49)
[20:46:31][Gradle failure report] 	at org.gradle.internal.fingerprint.impl.AbstractFileCollectionFingerprinter.fingerprint(AbstractFileCollectionFingerprinter.java:46)
[20:46:31][Gradle failure report] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.lambda$visitInputFileProperties$1(ExecuteActionsTaskExecuter.java:324)
[20:46:31][Gradle failure report] 	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.lambda$fingerprintInputFiles$3(CaptureStateBeforeExecutionStep.java:192)
[20:46:31][Gradle failure report] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.visitInputFileProperties(ExecuteActionsTaskExecuter.java:322)
[20:46:31][Gradle failure report] 	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.fingerprintInputFiles(CaptureStateBeforeExecutionStep.java:188)
[20:46:31][Gradle failure report] 	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.captureExecutionState(CaptureStateBeforeExecutionStep.java:150)
[20:46:31][Gradle failure report] 	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.lambda$captureExecutionStateOp$1(CaptureStateBeforeExecutionStep.java:104)
[20:46:31][Gradle failure report] 	at org.gradle.internal.execution.steps.BuildOperationStep$1.call(BuildOperationStep.java:40)
[20:46:31][Gradle failure report] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
[20:46:31][Gradle failure report] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
[20:46:31][Gradle failure report] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
[20:46:31][Gradle failure report] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
[20:46:31][Gradle failure report] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
[20:46:31][Gradle failure report] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
[20:46:31][Gradle failure report] 	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
[20:46:31][Gradle failure report] 	at org.gradle.internal.execution.steps.BuildOperationStep.operation(BuildOperationStep.java:37)
[20:46:31][Gradle failure report] 	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.captureExecutionStateOp(CaptureStateBeforeExecutionStep.java:103)
[20:46:31][Gradle failure report] 	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.lambda$execute$0(CaptureStateBeforeExecutionStep.java:78)
[20:46:31][Gradle failure report] 	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:78)
[20:46:31][Gradle failure report] 	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:53)
[20:46:31][Gradle failure report] 	at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:74)
[20:46:31][Gradle failure report] 	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.lambda$execute$2(SkipEmptyWorkStep.java:78)
[20:46:31][Gradle failure report] 	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:78)
[20:46:31][Gradle failure report] 	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:34)
[20:46:31][Gradle failure report] 	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:39)
[20:46:31][Gradle failure report] 	at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:40)
[20:46:31][Gradle failure report] 	at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:28)
[20:46:31][Gradle failure report] 	at org.gradle.internal.execution.impl.DefaultWorkExecutor.execute(DefaultWorkExecutor.java:33)
[20:46:31][Gradle failure report] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:192)
[20:46:31][Gradle failure report] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:184)
[20:46:31][Gradle failure report] 	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:114)
[20:46:31][Gradle failure report] 	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
[20:46:31][Gradle failure report] 	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:62)
[20:46:31][Gradle failure report] 	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
[20:46:31][Gradle failure report] 	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
[20:46:31][Gradle failure report] 	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
[20:46:31][Gradle failure report] 	... 23 more
```
</details>

Note this weird path value:
> Caused by: java.nio.file.InvalidPathException: Illegal char <:> at index 59: D:\BuildAgent\work\3534af43999e68d2\modules\intellij\task ':studio-intellij:jarSearchableOptions' property 'destinationDirectory'\task ':studio-intellij:jarSearchableOptions' property 'archiveFileName'

I've tried to launch build process in debug mode and debug the plugin.
Found that exception occurs when trying to calculate input files snapshot for the "buildPlugin" task.

Looks like task inputs for Gradle > 5.1 are invalid.
See this code:
org.jetbrains.intellij.IntelliJPlugin#configureBuildPluginTask

```
from({
    if (VersionNumber.parse(project.gradle.gradleVersion) >= VersionNumber.parse("5.1")) {
        "${jarSearchableOptionsTask.getDestinationDirectory()}/${jarSearchableOptionsTask.getArchiveFileName()}"
    } else {
...                }
}) { into 'lib' }
```
But DirectoryProperty#toString() does NOT return directory path. It returns meta information about this property. See screenshot from the debugger:
![image](https://user-images.githubusercontent.com/6158807/81693751-68483780-9471-11ea-8d0a-09f26712aafc.png)

Not being an expert in Gradle, I can't suggest a fix. Maybe call DirectoryProperty#get() instead?

**Comment by zolotov** [comment]
thanks! will be fixed in 0.4.21

**Comment by alexbudarov** [comment]
Hi,
Thank you for your prompt reaction.
Just wanted to note that nearby
> jarSearchableOptionsTask.getArchiveFileName()

also returns a property, not a String.

**Comment by zolotov** [comment]
you're right, thanks

**Comment by zolotov** [comment]
published 0.4.21

**Comment by alexbudarov** [comment]
I've tried 0.4.21, and the problem is gone. Thanks.
